xdsl/dialects/llvm.py

```diff
@@ -134,29 +132,20 @@ def get(ptr: SSAValue | Operation,
         # convert a potential Operation into an SSAValue
         ptr = SSAValue.get(ptr)
 
-        if ptr_type is None:
-            ptr_type = LLVMPointerType.opaque()
-        else:
-            ptr_type = LLVMPointerType.typed(ptr_type)
+        if not isinstance(result_type, LLVMPointerType):
+            raise ValueError('...')
 
-        # if no result type was give, infer from pointer type
-        if result_type is None:
-            # ptr is an SSAValue
-            # ptr.typ is the type of the SSAValue (so the LLVMPointerType)
-            assert isinstance(ptr.typ, LLVMPointerType)
-            # ptr.typ.type is the wrapped type of the pointer
-            # use that as the result type
-            result_type = ptr_type
+        if not ptr.typ.is_typed():  #type: ignore
+            print('hello')
+            if pointee_type == None:
+                raise ValueError('...')
 
         attrs: dict[str, Attribute] = {
             'rawConstantIndices': indices_attr,
-            'elem_type': result_type
         }
 
-        if ptr_type.is_typed():
-            attrs.pop('elem_type')
-
-        result_type = LLVMPointerType.typed(result_type)
+        if not ptr.typ.is_typed():  #type: ignore
```

Comment:
Why the type ignore?

Reply:
Pyright keeps trying to access .is_typed() as a method within typ. Could probably be fixed if we pass ptr.typ as an argument instead of self if this is an issue

Reply:
I'm not sure what you mean by "Pyright keeps trying to access .is_typed() as a method within typ". How is Pyright accessing a method?